src/main/scala/nodes/images/Pooler.scala

```diff
@@ -67,3 +67,21 @@ class Pooler(
     ChannelMajorArrayVectorizedImage(patch, ImageMetadata(numPoolsX, numPoolsY, numChannels))
   }
 }
+
+object Pooler {
+
+  def sumPooler(in: DenseVector[Double]): Double = {
```

Comment:
I see the same thing - Not sure why the while loop might be slower, but I feel this isn't worth investigating. Changed it to just use `sum(_)` now